prime-router/docs/onboarding-users/transport/rest.md

```diff
@@ -0,0 +1,160 @@
+## 1.) How ReportStream REST stransport works
+When STLT message is ready to be sent, ReportStream uploads the message to STLT's ready queue folder found in Azure Blobstorage (e.g., la-phd.elr, ca-doh.elr, etc.) and then triggers the send queue to start the SendFunction.
+<ol type="a">
+    <li>Using receiver.TransportType, the SendFunction calls to specific transport service accordingly. In our case, the SendFunction calls to the RESTTransport service.</li>
+    <li>Reading credential from Azure Vault, RESTTransport uses its Type to determine the authentication type (i.e., UserPass or UserApiKey).  Next, it uses the obtained key to authenticates with STLT and sends the message payload.</li>
+</ol>
+
+## 2.) How-to prepare a key for authentication
+For authentication, you need to do both step a and step b.
+<ol type="a">
+<li>Generate the "Credential in JSON format" for authentication</li>
+
+Currently, RESTTransport uses the one of the three options: i) UserPass, ii) UserApiKey with JKS, or iii) UserApiKey with two-legged credential type to authenticate and obtain Bearer token from STLT. User can use **primeCLI** command with credential-create option to generate the "Credential in JSON format" as given below.
+
+- With STLT's credential username and password given to us by STLT, user need to run the following command to generate the UserPass credential type object:
+
+        Command:
+            ./prime credential-create --type UserPass --user <username> --pass <password>
+        
+        Output: 
+            Credential in JSON format: 
+            ===
+            {"@type":"UserPass","user":"<username>","pass":"<password>"}
+            ===
+
+- With STLT's credential username and API Key given to us by STLT, user need to run the following command to generate the UserApiKey credential type object:
+	
+          Command:
+              ./prime credential-create --type UserApiKey --apikey-user <username> --apikey <API Key>
+        
+          Output: 
+              Credential in JSON format: 
+              ===
+              {"@type":"UserApiKey","user":"<username>","key":"<API Key>"}
+              ===
+
+- With STLT's credential jks-Username, jks-File-Password, and jks-key-file given to us by STLT, user need to run the following command to generate the UserJks credential type object:
+	
+        Command:
+            ./prime credential-create --type UserJks --jks-use <JKS Username> --jks-file-pass <JKS File Password> --jks-file <PATH to JKS File>
+            
+        Output: 
+            Credential in JSON format: 
+            ===
+            {"@type":"UserJks","user":"<JKS Username>", "jks":"<JKS Key>"}
+            ===
+
+<li>Create Azure Key Vault Secret for STLT</li>
+
+With CDC SU password, login into the **Microsoft Azure** and navigate to the **pdhprod-clientconfig** and select **Secrets** from the menu on the left.  You should see the following screenshot.  **Note: Make sure your prod VPN is on**.
```

Comment:
Is VPN still necessary/does it still work with this? I swear I've done it without, and I know there have been some changes made regarding the VPN since I started.

Reply:
you shouldn't need VPN if your IP is whitelisted in pdhprod-clientconfig. Currently we only need VPN to access the staging DB (no idea why that's the line)

Reply:
Can you please correct the wording as you see fit. I'm not sure how I would best update that.

Reply:
```suggestion
With CDC SU password, login into the **Microsoft Azure** and navigate to the **pdhprod-clientconfig** and select **Secrets** from the menu on the left.  You should see the following screenshot.
```

Reply:
Yes, for production, we still need to turn on VPN.